datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/InboxController.java

```diff
@@ -70,6 +91,8 @@ public void readFromInbox(@RequestHeader String user,
              OutputStream os = response.getOutputStream()
         ) {
             StreamUtils.copy(is, os);
+        } catch (Exception e) {
+            throw new FileNotFoundException(String.format("File '%s' not found for user '%s'", path, user), e);
```

Comment:
It is very generic wrapping - there can be credentials issues causing exception

Reply:
I tried use exception which actually raised when password wrong - UnrecoverableKeyException, but compiler complaining that this exception is never thrown. It's strange

Reply:
You need to do 
```
catch (Exception e) {
if (e instanceof UnrecoverableKeyException) {
throw new BadCredentialsException("");
}
}
```

Reply:
done in controller advice